### PR TITLE
Move sam template inspection into sam template utils

### PR DESCRIPTION
**Pull request by abrooksv**
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtilsTest.kt [hunk 1]
```diff
@@ -27,10 +31,178 @@ class SamTemplateUtilsTest {
...
+    fun `wrong S3 scheme throws`() {
+        val templateFile = tempFolder.newFile().toPath()
+        templateFile.writeText(
+            """
+            Resources:
+              Function:
+                Type: AWS::Serverless::Function
+                Properties:
+                  Handler: helloworld.App::handleRequest
+                  CodeUri:  file://FooBucket/Foo/Key
```
Comment (hunterwerlla) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtilsTest.kt:148]: nit: extra space

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtilsTest.kt [hunk 2]
```diff
@@ -27,10 +31,178 @@ class SamTemplateUtilsTest {
...
+        val uploadedCode = SamTemplateUtils.getUploadedCodeUri(templateFile, "Function")
+        assertThat(uploadedCode).isInstanceOfSatisfying(UploadedS3Code::class.java) { code ->
+            assertThat(code.bucket).isEqualTo("FooBucket")
+            assertThat(code.key).isEqualTo("FooKey")
+            assertThat(code.version).isEqualTo("FooVersion")
+        }
+    }
+
+    @Test
+    fun `no package type is treated as Zip`() {
```
Comment (hunterwerlla) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtilsTest.kt:114]: Is there a test in here with image?

Comment (abrooksv) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtilsTest.kt:114]: Image URI parsing tests images

Comment (abrooksv) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtilsTest.kt:114]: `image uri block is parsed`

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtilsTest.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtilsTest.kt]: fixed

**Review by hunterwerlla — APPROVED** [review]

**Pull request merged**